packages/sdk/src/index.ts

```diff
@@ -3,35 +3,64 @@ import {
   CredbullFixedYieldVault__factory,
   ERC20__factory,
 } from '@credbull/contracts';
+import type { CredbullFixedYieldVault, CredbullFixedYieldVaultWithUpside, ERC20 } from '@credbull/contracts';
 import { BigNumber, Signer } from 'ethers';
 import { ethers } from 'ethers';
 import { SiweMessage, generateNonce } from 'siwe';
 
-import { decodeContractError } from './src/utils/utils';
+import { decodeContractError } from './utils';
 
+/**
+ * The Credbull SDK for creating and managing Credbull Vaults, via the Credbull API.
+ */
 export class CredbullSDK {
-  private SERVICE_URL = process.env.API_BASE_URL || '';
+  // The `URL` for the Credbull API.
+  private serviceUrl: URL;
+
+  // The logged in User's Access Token. Only populated on login.
+  private accessToken: string | undefined = undefined;
+
+  /**
+   * Creates a `CredbullSDK` instance pointed at the `serviceUrl` API backend.
+   *
+   * @param serviceUrl The `string` API URL. Must be a valid URL.
+   * @param email The User's Email Address.
+   * @param password The User's Password.
+   * @param signer The `Signer`.
+   * @throws TypeError if `serviceUrl` is not a valid URL.
+   */
   constructor(
```

Comment:
### Credentials
My original take is that the SDK is to encapsulate the API, so having the Access Token, means the client has already used the API to log in. Which seems against the function of the SDK. 
 However, if the Access Token is already present, then we should not force the client to re-log in. I will change the SDK to have a unified type constructor parameter, allowing the passing of username/password or accessToken. That resolves all issues.

### Signer
This is needed for all direct (JSON RPC) calls to the blockchain layer. However, I agree that IF the SDK encapsulates the API, then all operations should be via the API. Nice.

Reply:
Jonathan and i had a great philosophical discussion about who owns the vaults, us or the DaaS clients.  it is us.  that means that smart contract operations will be Credbull wallets.  so, we need to split the user/pass and signer most likely as user/pass is DaaS, and signer is us.  

also, connecting wallets is a natural interaction pattern for dApps.  does the DaaS client even need a back-end version of the SDK, or should everything go through a Web UI?

- [ ] Look at how Safe or other SDK's connect / login - Ian
- [ ] Does ether.js have ways to connect without a private key (ala Alchemy APIs).  Then what we need is the DaaS clients to sign up for an Ethers API key and provide that, rather than a private key.  - Jonathan
- [ ] Check with Pedro for SDK interaction pattern: is it Web based on back-end or no preference - Jonathan

Reply:
for the first bullet point, confirmed that Safe uses the Private Key as well in their SDK (e.g. https://docs.safe.global/sdk/protocol-kit).  So i'm fine with keeping to it as a requirement.  Now it's more about how to protect the keys we need to provide.